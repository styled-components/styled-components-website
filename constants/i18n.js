const DEFAULT_TRANSLATION = 'translations'
const NAV_TRANSLATION = 'nav'
const HOME_TRANSLATION = 'home'
const DOCS_TRANSLATION = 'docs'
const DOCS_BASICS_TRANSLATION = 'docsBasics'
const DOCS_ADVANCED_TRANSLATION = 'docsAdvanced'
const DOCS_API_TRANSLATION = 'docsApi'
const DOCS_FAQS_TRANSLATION = 'docsFaqs'

const TRANSLATIONS = [
  DEFAULT_TRANSLATION,
  NAV_TRANSLATION,
  HOME_TRANSLATION,
  DOCS_TRANSLATION,
  DOCS_BASICS_TRANSLATION,
  DOCS_ADVANCED_TRANSLATION,
  DOCS_API_TRANSLATION,
  DOCS_FAQS_TRANSLATION,
]

const DEFAULT_LANGUAGE = 'en'

const LANGUAGES = [
  DEFAULT_LANGUAGE,
  'es',
]

module.exports = {
  DEFAULT_TRANSLATION,
  NAV_TRANSLATION,
  TRANSLATIONS,
  HOME_TRANSLATION,
  DOCS_TRANSLATION,
  DOCS_BASICS_TRANSLATION,
  DOCS_ADVANCED_TRANSLATION,
  DOCS_API_TRANSLATION,
  DOCS_FAQS_TRANSLATION,
  LANGUAGES,
  DEFAULT_LANGUAGE,
}
