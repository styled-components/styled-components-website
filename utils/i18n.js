import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { reactI18nextModule } from 'react-i18next'

import translations from '../locales'

i18n
  .use(LanguageDetector)
  .use(reactI18nextModule)
  .init({
    fallbackLng: 'en',

    // have a common namespace used around the full app
    ns: ['translations', 'homepage'],
    defaultNS: 'translations',

    debug: process.env.NODE_ENV !== 'production',

    interpolation: {
      escapeValue: false, // not needed for react!!
    },

    react: {
      wait: true
    },

    resources: { ...translations }
  })

export default i18n
