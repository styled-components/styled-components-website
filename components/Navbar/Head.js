import styled, { css } from 'styled-components'

import rem from '../../utils/rem'
import { mobile } from '../../utils/media'

const Head = styled.div`
  height: ${rem(300)};
  padding: ${rem(65)} ${rem(85)} ${rem(45)};

  ${mobile(css`
    display: flex;
    flex-direction: row;
    align-items: center;

    height: ${rem(70)};
    padding: 0 ${rem(15)};
  `)}
`

export default Head
