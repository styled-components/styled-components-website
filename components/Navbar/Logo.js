import styled, { css } from 'styled-components'

import rem from '../../utils/rem'
import { mobile } from '../../utils/media'

const Logo = styled.div`
  display: inline-block;
  width: 100%;
  height: 100%;

  box-sizing: border-box;

  background-image: url(/static/logo.png);
  background-size: contain;
  background-position: center;

  ${mobile(css`
    background-image: url(/static/icon.png);
    background-position: center;
    width: ${rem(70)};
    height: ${rem(70)};
  `)}
`

export default Logo
