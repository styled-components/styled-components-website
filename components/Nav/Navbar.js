import React, { PureComponent, createRef } from 'react';
import styled, { css } from 'styled-components';
import { Close } from '@styled-icons/material';
import rem from '../../utils/rem';
import { blmGrey, paleGrey } from '../../utils/colors';
import { navbarHeight } from '../../utils/sizes';
import { headerFont } from '../../utils/fonts';
import { mobile } from '../../utils/media';
import Link from '../Link';
import NavLinks from './NavLinks';
import Social from './Social';
import Logo from './Logo';
import MobileNavbar from './MobileNavbar';
import SearchWithAlgolia from './SearchWithAlgolia';

const Wrapper = styled.nav`
  position: fixed;
  left: 0;
  box-sizing: border-box;
  z-index: 3;

  width: 100%;
  height: ${rem(navbarHeight)};

  font-family: ${headerFont};
  font-size: ${rem(15)};
  font-weight: 500;
  background: ${(props) => (props.transparent ? 'transparent' : blmGrey)};
  transition: background 300ms ease-out;
  color: white;
  padding: 0;
`;

const StartWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
`;

const EndWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
`;
/* stylelint-disable */
const StyledSocial = styled(Social)``;
/* stylelint-enable */

const NormalNavbar = styled.div`
  display: flex;
  align-items: center;
  padding: 0 ${rem(20)};
  justify-content: space-between;
  ${StartWrapper}, ${EndWrapper} ${StyledSocial} {
    ${mobile(css`
      display: none;
    `)};
  }
`;

const StyledModalCloseIcon = styled(Close)`
  width: ${rem(26)};
  height: ${rem(26)};
  color: white;
`;

const AlgoliaModal = styled.div`
  ${mobile(css`
    background: currentColor;
    overflow: auto;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
  `)};
`;

const baseZ = 1;

const AlgoliaModalHeader = styled.div`
  display: none;
  color: currentColor;

  ${mobile(css`
    display: ${(props) => (props.isOpen ? 'block' : 'none')};

    button {
      cursor: pointer;
      padding: 0;
      position: fixed;
      right: ${rem(10)};
      top: ${rem(11)};
      border: none;
      z-index: ${baseZ + 1};
    }
  `)};
`;

const AlgoliaModalOverlay = styled.div`
  margin-right: ${rem(10)};

  ${mobile(css`
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: ${baseZ};
    left: 0;
    display: ${(props) => (props.isOpen ? 'block' : 'none')};
    background: ${paleGrey};
    overflow-y: auto;
    margin: 0;

    .algolia-autocomplete .ds-dropdown-menu {
      position: static !important;
      display: block;
      max-width: 100%;
      min-width: 0;
    }
  `)};
`;

class ModalContainer extends PureComponent {
  modalElement = createRef();
  onModalOverlayClick = (e) => {
    if (!this.modalElement.current.contains(e.target)) {
      this.props.requestModalClose();
    }
  };
  onCloseButtonClick = (e) => {
    e.stopPropagation();
    this.props.requestModalClose();
  };
  render() {
    return (
      <>
        <AlgoliaModalHeader isOpen={this.props.isOpen}>
          <button onClick={this.onCloseButtonClick}>
            <StyledModalCloseIcon />
          </button>
        </AlgoliaModalHeader>
        <AlgoliaModalOverlay onClick={this.onModalOverlayClick} isOpen={this.props.isOpen}>
          <AlgoliaModal ref={this.modalElement}>
            <div>{this.props.children}</div>
          </AlgoliaModal>
        </AlgoliaModalOverlay>
      </>
    );
  }
}

const LogoLink = styled(Link).attrs((/* props */) => ({
  unstyled: true,
  href: '/',
}))`
  display: inline-block;
  vertical-align: center;
  margin-right: ${rem(35)};
`;

class Navbar extends PureComponent {
  state = {
    isOpen: false,
  };
  openModal = () => this.setState(() => ({ isOpen: true }));
  closeModal = () => this.setState(({ isOpen }) => (isOpen ? { isOpen: false } : null));
  render() {
    const { onSideToggle, onMobileNavToggle, isSideFolded, isMobileNavFolded, showSideNav, transparent } = this.props;

    return (
      <Wrapper transparent={transparent}>
        <NormalNavbar>
          <StartWrapper>
            <LogoLink>
              <Logo />
            </LogoLink>
            <NavLinks />
          </StartWrapper>
          <EndWrapper>
            <ModalContainer isOpen={this.state.isOpen} requestModalClose={this.closeModal}>
              <SearchWithAlgolia requestModalClose={this.closeModal} />
            </ModalContainer>
            <StyledSocial />
          </EndWrapper>
        </NormalNavbar>
        <MobileNavbar
          isSideFolded={isSideFolded}
          onSearchButtonClick={this.openModal}
          isMobileNavFolded={isMobileNavFolded}
          onSideToggle={onSideToggle}
          onMobileNavToggle={onMobileNavToggle}
          showSideNav={showSideNav}
        />
      </Wrapper>
    );
  }
}

export default Navbar;
