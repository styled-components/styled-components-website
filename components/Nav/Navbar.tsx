import { Close } from '@styled-icons/material';
import { PureComponent, createRef } from 'react';
import styled, { css } from 'styled-components';
import { blmGrey, paleGrey } from '../../utils/colors';
import { headerFont } from '../../utils/fonts';
import { mobile } from '../../utils/media';
import rem from '../../utils/rem';
import { navbarHeight } from '../../utils/sizes';
import Link from '../Link';
import Logo from './Logo';
import MobileNavbar from './MobileNavbar';
import NavLinks from './NavLinks';
import SearchWithAlgolia from './SearchWithAlgolia';
import Social from './Social';

const Wrapper = styled.nav<{ $transparent?: boolean }>`
  align-items: center;
  background-color: rgba(12, 13, 15, 0.7);
  backdrop-filter: blur(5px);
  box-sizing: border-box;
  color: white;
  display: flex;
  flex-wrap: wrap;
  font-family: ${headerFont};
  font-size: ${rem(15)};
  font-weight: 500;
  justify-content: center;
  height: ${rem(navbarHeight)};
  left: 0;
  padding: 0;
  position: fixed;
  transition: background 300ms ease-out;
  width: 100%;
  z-index: 3;
`;

const StartWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
`;

const EndWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
`;
/* stylelint-disable */
const StyledSocial = styled(Social)``;
/* stylelint-enable */

const NormalNavbar = styled.div`
  display: flex;
  align-items: center;
  padding: 0 ${rem(20)};
  justify-content: space-between;

  ${StartWrapper}, ${EndWrapper} ${StyledSocial} {
    ${mobile(css`
      display: none;
    `)};
  }
`;

const StyledModalCloseIcon = styled(Close)`
  width: ${rem(26)};
  height: ${rem(26)};
  color: white;
`;

const AlgoliaModal = styled.div`
  ${mobile(css`
    background: currentColor;
    overflow: auto;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
  `)};
`;

const baseZ = 1;

const AlgoliaModalHeader = styled.div<{ $isOpen?: boolean }>`
  display: none;
  color: currentColor;

  ${mobile(css<{ $isOpen?: boolean }>`
    display: ${props => (props.$isOpen ? 'block' : 'none')};

    button {
      cursor: pointer;
      padding: 0;
      position: fixed;
      right: ${rem(10)};
      top: ${rem(11)};
      border: none;
      z-index: ${baseZ + 1};
    }
  `)};
`;

const AlgoliaModalOverlay = styled.div<{ $isOpen?: boolean }>`
  margin-right: ${rem(10)};

  ${mobile(css<{ $isOpen?: boolean }>`
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: ${baseZ};
    left: 0;
    display: ${props => (props.$isOpen ? 'block' : 'none')};
    background: ${paleGrey};
    overflow-y: auto;
    margin: 0;

    .algolia-autocomplete .ds-dropdown-menu {
      position: static !important;
      display: block;
      max-width: 100%;
      min-width: 0;
    }
  `)};
`;

class ModalContainer extends PureComponent<{ isOpen?: boolean; requestModalClose: () => void }> {
  modalElement = createRef<HTMLDivElement>();

  onModalOverlayClick = (e: React.MouseEvent) => {
    if (!this.modalElement.current?.contains(e.target as Node)) {
      this.props.requestModalClose();
    }
  };

  onCloseButtonClick = (e: React.MouseEvent) => {
    e.stopPropagation();
    this.props.requestModalClose();
  };

  render() {
    return (
      <>
        <AlgoliaModalHeader $isOpen={this.props.isOpen}>
          <button onClick={this.onCloseButtonClick}>
            <StyledModalCloseIcon />
          </button>
        </AlgoliaModalHeader>
        <AlgoliaModalOverlay onClick={this.onModalOverlayClick} $isOpen={this.props.isOpen}>
          <AlgoliaModal ref={this.modalElement}>
            <div>{this.props.children}</div>
          </AlgoliaModal>
        </AlgoliaModalOverlay>
      </>
    );
  }
}

const LogoLink = styled(Link).attrs((/* props */) => ({
  unstyled: true,
  href: '/',
}))`
  display: inline-block;
  vertical-align: center;
  margin-right: ${rem(35)};
`;

export interface NavbarProps {
  onSideToggle?: () => void;
  onMobileNavToggle?: () => void;
  isSideFolded?: boolean;
  isMobileNavFolded?: boolean;
  showSideNav?: boolean;
}

function Navbar({ onSideToggle, onMobileNavToggle, isSideFolded, isMobileNavFolded, showSideNav }: NavbarProps) {
  return (
    <Wrapper>
      <MobileNavbar
        isSideFolded={isSideFolded}
        isMobileNavFolded={isMobileNavFolded}
        onSideToggle={onSideToggle}
        onMobileNavToggle={onMobileNavToggle}
        showSideNav={showSideNav}
      />

      <NormalNavbar>
        <StartWrapper>
          <LogoLink aria-label="Styled-components logo">
            <Logo />
          </LogoLink>
          <NavLinks />
        </StartWrapper>
      </NormalNavbar>

      <EndWrapper
        css={css`
          margin-left: auto;
          margin-right: 16px;

          ${mobile(css`
            margin-right: 48px;
          `)}
        `}
      >
        <SearchWithAlgolia />
        <StyledSocial style={{ marginLeft: 16 }} />
      </EndWrapper>
    </Wrapper>
  );
}

export default Navbar;
