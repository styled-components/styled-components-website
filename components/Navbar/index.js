import React, { Component } from 'react'
import Sidebar from './Sidebar'
import Head from './Head'
import Logo from './Logo'
import Text from './Text'
import Menu from './Menu'
import MenuButton from './MenuButton'
import Link from '../Link'

class Navbar extends Component {
  state = {
    isFolded: true
  }

  onFold = () => {
    this.setState({
      isFolded: !this.state.isFolded
    })
  }

  onRouteChange = () => {
    this.setState({ isFolded: true })
  }

  render() {
    const { isFolded } = this.state

    return (
      <Sidebar>
        <Head>
          <Link unstyled href="/docs">
            <Logo />
          </Link>

          <Text>
            <Link unstyled href="/docs">
              styled-components
            </Link>
          </Text>

          <MenuButton onClick={this.onFold} />
        </Head>

        <Menu
          isFolded={isFolded}
          onRouteChange={this.onRouteChange}
        />
      </Sidebar>
    )
  }
}

export default Navbar
