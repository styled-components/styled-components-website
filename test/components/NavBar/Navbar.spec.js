import React from 'react'
import { mount } from 'enzyme'
import 'jest-styled-components'
import { I18nextProvider } from 'react-i18next'
import Navbar from '../../../components/Nav/Navbar'

import i18n from '../../helpers/i18n'

const wrapper = (language, props) => mount(
  <I18nextProvider i18n={i18n} initialLanguage={language}>
    <Navbar />
  </I18nextProvider>
)

test('Navbar renders correctly', () => {
  const component = wrapper('en')
  expect(component.find(Navbar)).toMatchSnapshot()
})
