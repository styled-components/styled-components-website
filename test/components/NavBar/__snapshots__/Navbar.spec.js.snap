// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
.c7 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 0.2777777777777778rem;
  height: 0.2777777777777778rem;
  margin: 0 0.8333333333333334rem;
  border-radius: 50%;
  background: currentColor;
  opacity: 0.35;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 1.6666666666666667rem;
}

.c6 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: inline-block;
  line-height: 2.7777777777777777rem;
  -webkit-transition: opacity 0.2s,-webkit-transform 0.2s;
  -webkit-transition: opacity 0.2s,transform 0.2s;
  transition: opacity 0.2s,transform 0.2s;
  cursor: pointer;
  -webkit-letter-spacing: 0.022222222222222223rem;
  -moz-letter-spacing: 0.022222222222222223rem;
  -ms-letter-spacing: 0.022222222222222223rem;
  letter-spacing: 0.022222222222222223rem;
  color: currentColor;
}

.c6:hover,
.c6:focus {
  opacity: 0.8;
}

.c6:active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
  opacity: 0.6;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: 1.1111111111111112rem;
  line-height: 2.7777777777777777rem;
  -webkit-transition: opacity 0.2s,-webkit-transform 0.2s;
  -webkit-transition: opacity 0.2s,transform 0.2s;
  transition: opacity 0.2s,transform 0.2s;
  cursor: pointer;
}

.c17:last-child {
  margin-right: 0;
}

.c17:hover,
.c17:focus {
  opacity: 0.8;
}

.c17:active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
  opacity: 0.6;
}

.c17 svg path {
  fill: currentColor;
}

.c18 {
  width: 0.7777777777777778rem;
  height: 0.7777777777777778rem;
}

.c19.c19 {
  width: NaNrem;
  height: 1rem;
}

.c4 {
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;
  background-position: center;
  background-size: contain;
  background-image: url(/nav-logo.png);
  width: 9.11111111111111rem;
  height: 2.2222222222222223rem;
}

.c25 {
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;
  background-position: center;
  background-size: contain;
  background-image: url(/icon.png);
  width: 5.555555555555555rem;
  height: 1.6666666666666667rem;
}

.c23 svg {
  display: inline-block;
}

.c23 svg path {
  fill: currentColor;
}

.c22 {
  background: none;
  outline: none;
  border: none;
  background: rgba(0,0,0,0.07);
  cursor: pointer;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 2.7777777777777777rem;
  padding: 0 0.5555555555555556rem;
  text-align: center;
  vertical-align: middle;
}

.c21 + .c21 {
  padding-left: 0;
}

.c26 {
  background: none;
  outline: none;
  border: none;
  background: none;
  cursor: pointer;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 2.7777777777777777rem;
  padding: 0 0.5555555555555556rem;
  text-align: center;
  vertical-align: middle;
}

.c21 + .c21 {
  padding-left: 0;
}

.c20 {
  display: none;
}

.c30 {
  position: absolute;
  top: 2.7777777777777777rem;
  left: 0;
  right: 0;
  height: 2.7777777777777777rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 1.1111111111111112rem;
  -webkit-transition: height 0.1s;
  transition: height 0.1s;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-overflow-scrolling: touch;
  overflow-x: scroll;
  overflow-y: hidden;
  background: #efefef;
  color: #868686;
}

.c24 {
  display: inline-block;
  vertical-align: center;
}

.c28 {
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c31 {
  padding-right: 1.1111111111111112rem;
}

.c27.c27 {
  width: 1.5555555555555556rem;
  height: 1.5555555555555556rem;
}

.c29.c29 {
  width: 2rem;
  height: 2rem;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c15 {
  background: none;
  outline: none;
  border: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 7.222222222222222rem;
  line-height: 1.6666666666666667rem;
  font-size: 0.8333333333333334rem;
  color: currentColor;
}

.c15::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
}

.c15::-moz-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
}

.c15:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
}

.c15::placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
}

.c15:focus::-webkit-input-placeholder {
  opacity: 0.8;
}

.c15:focus::-moz-placeholder {
  opacity: 0.8;
}

.c15:focus:-ms-input-placeholder {
  opacity: 0.8;
}

.c15:focus::placeholder {
  opacity: 0.8;
}

.c14 {
  background: none;
  outline: none;
  border: none;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  background: none;
  outline: none;
  border: none;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 0.2222222222222222rem;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c14:hover,
.c14:focus {
  opacity: 0.7;
  -webkit-filter: drop-shadow(0 1px 2px rgba(0,0,0,0.2));
  filter: drop-shadow(0 1px 2px rgba(0,0,0,0.2));
}

.c14:active {
  background: rgba(0,0,0,0.1);
}

.c0 {
  position: fixed;
  left: 0;
  box-sizing: border-box;
  z-index: 3;
  width: 100%;
  height: 2.7777777777777777rem;
  font-family: "Avenir Next",-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8333333333333334rem;
  font-weight: 500;
  background: rgb(33,33,33);
  -webkit-transition: background 300ms ease-out;
  transition: background 300ms ease-out;
  color: white;
  padding: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1.1111111111111112rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c10 {
  width: 1.4444444444444444rem;
  height: 1.4444444444444444rem;
  color: white;
}

.c9 {
  display: none;
  color: currentColor;
}

.c11 {
  margin-right: 0.5555555555555556rem;
}

.c3 {
  display: inline-block;
  vertical-align: center;
  margin-right: 1.9444444444444444rem;
}

@media (max-width:62.5em) {
  .c20 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    height: 2.7777777777777777rem;
  }
}

@media (max-width:62.5em) {
  .c13 {
    display: block;
    padding: 0;
  }

  .c13 span.algolia-autocomplete {
    display: block !important;
  }

  .c13 span.ds-dropdown-menu::before {
    display: none;
  }
}

@media (max-width:62.5em) {
  .c15 {
    padding: 0.5555555555555556rem 2.6666666666666665rem;
    display: block;
    width: 100%;
    background: rgb(33,33,33);
    color: white;
  }
}

@media (max-width:62.5em) {
  .c14 {
    position: absolute;
    top: 0.5rem;
    left: 0.5555555555555556rem;
    height: 1.7777777777777777rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    width: 1.7777777777777777rem;
    z-index: 1;
    background: none;
    color: white;
  }
}

@media (max-width:62.5em) {

}

@media (max-width:62.5em) {
  .c12 {
    background: currentColor;
    overflow: auto;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
  }
}

@media (max-width:62.5em) {
  .c9 {
    display: none;
  }

  .c9 button {
    cursor: pointer;
    padding: 0;
    position: fixed;
    right: 0.5555555555555556rem;
    top: 0.6111111111111112rem;
    border: none;
    z-index: 2;
  }
}

@media (max-width:62.5em) {
  .c11 {
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: 1;
    left: 0;
    display: none;
    background: #efefef;
    overflow-y: auto;
    margin: 0;
  }

  .c11 .algolia-autocomplete .ds-dropdown-menu {
    position: static !important;
    display: block;
    max-width: 100%;
    min-width: 0;
  }
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <a
        className="c3"
        href="/"
      >
        <div
          className="c4"
        />
      </a>
      <nav
        className="c5"
      >
        <a
          className="c6"
          href="/docs"
        >
          Documentation
        </a>
        <span
          className="c7"
        />
        <a
          className="c6"
          href="/showcase"
        >
          Showcase
        </a>
        <span
          className="c7"
        />
        <a
          className="c6"
          href="/ecosystem"
        >
          Ecosystem
        </a>
        <span
          className="c7"
        />
        <a
          className="c6"
          href="/releases"
        >
          Releases
        </a>
      </nav>
    </div>
    <div
      className="c8"
    >
      <div
        className="c9"
      >
        <button
          onClick={[Function]}
        >
          <svg
            className="c10"
          />
        </button>
      </div>
      <div
        className="c11"
        onClick={[Function]}
      >
        <div
          className="c12"
        >
          <div>
            <form
              className="c13"
            >
              <label
                className="c14"
                htmlFor="docs-search-input"
              >
                <svg
                  className=""
                />
              </label>
              <input
                className="c15"
                id="docs-search-input"
                placeholder="Search docs ..."
                type="search"
              />
            </form>
          </div>
        </div>
      </div>
      <nav
        className="c16 "
      >
        <a
          className="c17"
          href="https://spectrum.chat/styled-components/"
        >
          <svg
            className="c18"
            height="14"
            viewBox="0 0 15 15"
            width="14"
          >
            <title>
              spectrum
            </title>
            <path
              d="M0 6.5V1c0-.6.4-1 1-1 9 .3 13.7 5 14 14 0 .6-.4 1-1 1H8.5c-.6 0-1-.4-1-1-.3-4.4-2-6.2-6.5-6.5-.6 0-1-.4-1-1z"
              fill="#FFF"
            />
          </svg>
        </a>
        <a
          className="c17"
          href="https://github.com/styled-components"
        >
          <svg
            className="c19"
            height="18"
          />
        </a>
        <a
          className="c17"
          href="https://medium.com/styled-components"
        >
          <svg
            className="c19"
            height="18"
          />
        </a>
      </nav>
    </div>
  </div>
  <div
    className="c20"
  >
    <button
      className="c21 c22"
    >
      <svg
        className="c23"
        height="15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <title>
          close
        </title>
        <use
          fill="#FFF"
          transform="translate(1 1)"
          xlinkHref="#close"
        />
        <defs>
          <path
            d="M-.7.7l13 13 1.4-1.4-13-13L-.7.7zm13-1.4l-13 13 1.4 1.4 13-13-1.4-1.4z"
            id="close"
          />
        </defs>
      </svg>
    </button>
    <a
      aria-label="styled components"
      className="c24"
      href="/"
    >
      <div
        className="c25"
      />
    </a>
    <div>
      <button
        className="c21 c26"
        onClick={[Function]}
      >
        <svg
          className="c27"
          size={28}
        />
      </button>
      <button
        className="c21 c22"
      >
        <div
          className="c28"
        >
          <div
            className="c29"
            size={36}
          />
        </div>
      </button>
    </div>
    <div
      className="c30"
    >
      <nav
        className="c5"
      >
        <a
          className="c6"
          href="/docs"
        >
          Documentation
        </a>
        <span
          className="c7"
        />
        <a
          className="c6"
          href="/showcase"
        >
          Showcase
        </a>
        <span
          className="c7"
        />
        <a
          className="c6"
          href="/ecosystem"
        >
          Ecosystem
        </a>
        <span
          className="c7"
        />
        <a
          className="c6"
          href="/releases"
        >
          Releases
        </a>
      </nav>
      <div
        className="c31"
      >
        <nav
          className="c16"
        >
          <a
            className="c17"
            href="https://spectrum.chat/styled-components/"
          >
            <svg
              className="c18"
              height="14"
              viewBox="0 0 15 15"
              width="14"
            >
              <title>
                spectrum
              </title>
              <path
                d="M0 6.5V1c0-.6.4-1 1-1 9 .3 13.7 5 14 14 0 .6-.4 1-1 1H8.5c-.6 0-1-.4-1-1-.3-4.4-2-6.2-6.5-6.5-.6 0-1-.4-1-1z"
                fill="#FFF"
              />
            </svg>
          </a>
          <a
            className="c17"
            href="https://github.com/styled-components"
          >
            <svg
              className="c19"
              height="18"
            />
          </a>
          <a
            className="c17"
            href="https://medium.com/styled-components"
          >
            <svg
              className="c19"
              height="18"
            />
          </a>
        </nav>
      </div>
    </div>
  </div>
</nav>
`;
